File: ride-service/src/test/java/com/internship/rideservice/integration/BaseTest.java
package com.internship.rideservice.integration;

import com.internship.rideservice.RideServiceApplication;
import com.internship.rideservice.config.DisableCommunication;
import com.internship.rideservice.config.MongoDBContainerConfig;
import com.internship.rideservice.config.WireMockConfig;
import org.junit.jupiter.api.MethodOrderer;
import org.junit.jupiter.api.TestInstance;
import org.junit.jupiter.api.TestMethodOrder;
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.context.annotation.Import;
import org.springframework.test.context.ActiveProfiles;

@AutoConfigureMockMvc
@SpringBootTest(
        classes = RideServiceApplication.class,
        webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
@ActiveProfiles("test")
@Import({
        MongoDBContainerConfig.class,
        DisableCommunication.class,
        WireMockConfig.class
})
public abstract class BaseTest {

}

File: finance-service/src/test/java/com/internship/financeservice/integration/BaseTest.kt
package com.internship.financeservice.integration

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.databind.SerializationFeature
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
import com.fasterxml.jackson.module.kotlin.registerKotlinModule
import com.internship.financeservice.FinanceServiceApplication
import com.internship.financeservice.config.DisableBroker
import com.internship.financeservice.config.MySQLContainerConfig
import org.junit.jupiter.api.MethodOrderer
import org.junit.jupiter.api.TestInstance
import org.junit.jupiter.api.TestMethodOrder
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.context.annotation.Import
import org.springframework.test.context.ActiveProfiles
import org.springframework.test.web.servlet.MockMvc

@AutoConfigureMockMvc
@SpringBootTest(
    classes = [FinanceServiceApplication::class],
    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT
)
@ActiveProfiles("test")
@Import(
    MySQLContainerConfig::class,
    DisableBroker::class
)
abstract class BaseTest {

    @Autowired
    lateinit var mockMvc: MockMvc

    val objectMapper = ObjectMapper().apply {
        registerKotlinModule()
        registerModule(JavaTimeModule())
        disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS)
    }
}